blocks/blockstore/write_cache.go

```diff
@@ -8,7 +8,7 @@ import (
 )
 
 // WriteCached returns a blockstore that caches up to |size| unique writes (bs.Put).
-func WriteCached(bs Blockstore, size int) (Blockstore, error) {
+func WriteCached(bs Blockstore, size int) (*writecache, error) {
```

Comment:
Yes, this is the way we *should* be doing interfaces in go. that way, in the calling code, i can do either:
```
var bs Blockstore
bs = WriteCached()
```

OR

```
var gcbs GCBlockstore
gcbs = WriteCached()
```

without any ugly casts or failure conditions.